backend/mlarchive/archive/utils.py

```diff
@@ -372,3 +372,25 @@ def create_mbox_file(month, year, elist):
             msg = email.message_from_binary_file(f)
         mbox.add(msg)
     mbox.close()
+
+
+def update_mbox_files():
+    '''Update archive mbox files'''
+    yesterday = datetime.datetime.now() - timedelta(days=1)
+    month = yesterday.month
+    year = yesterday.year
+    for elist in EmailList.objects.filter(active=True, private=False):
+        if elist.message_set.filter(date__month=month, date__year=year).count() == 0:
+            continue
+        create_mbox_file(month=month, year=year, elist=elist)
```

Comment:
If I read the code correctly, `create_mbox_file()` will delete an existing mbox file before creating a new one. Is that desired / safe if the task is inadvertently run more than once on a given day? I.e., could data be written to the mbox created from the first execution, then lost when it's removed by the second?

Reply:
We write all messages for a month to the file each time, so data should not be lost. This is not as efficient as appending new messages to the file, but is necessary to account for removing messages (as intended) when they have been removed from the archive, like spam for instance, without having to perform a more complicated diff of content to specifically add or remove messages.

Reply:
Ah, I see. Thanks.